### PR TITLE
Fix sandbox being left enabled if an exception is thrown while rendering

### PR DESCRIPTION
**Pull request by CarsonF**
What's happening for me:
- An exception is thrown while rendering inside the sandbox
- Render exception response
- WebProfiler toolbar tries to include a file
- Sandbox is still enabled, so a SecurityException is thrown (which hides the real error)

### CODE REVIEW

**Comment by stof** [comment]
could you write a test covering this

File: lib/Twig/Extension/Core.php [hunk 1]
```diff
@@ -1402,6 +1402,12 @@ function twig_include(Twig_Environment $env, $context, $template, $variables = a
 
             throw $e;
         }
+    } catch (\Exception $e) {
```
Comment (CarsonF) [inline lib/Twig/Extension/Core.php]: Does PHP 5 choke on `catch (Throwable $e)`? 

Should that be before or after catching Exception? It probably doesn't matter in this case. I just remember reading somewhere that to make code compatible with PHP 5 `Throwable` should be caught first...

**Comment by CarsonF** [comment]
> could you write a test covering this ?

Since this involves the `include` function and the sandbox I'm not sure where I should add the test?
SandboxTest?

**Comment by fabpot** [comment]
@CarsonF Yes, `SandboxTest` seems to make more sense

**Comment by CarsonF** [comment]
@fabpot I don't think those CS changes should be made in this PR, right?

**Comment by fabpot** [comment]
Indeed, you can ignore them :)

File: lib/Twig/Extension/Core.php [hunk 2]
```diff
@@ -1406,6 +1406,18 @@ function twig_include(Twig_Environment $env, $context, $template, $variables = a
 
             throw $e;
         }
+    } catch (\Throwable $e) {
```
Comment (CarsonF) [inline lib/Twig/Extension/Core.php]: Ah, of course.

File: test/Twig/Tests/Extension/SandboxTest.php [hunk 3]
```diff
@@ -34,6 +34,7 @@ protected function setUp()
             '1_basic' => '{% if obj.foo %}{{ obj.foo|upper }}{% endif %}',
             '1_layout' => '{% block content %}{% endblock %}',
             '1_child' => "{% extends \"1_layout\" %}\n{% block content %}\n{{ \"a\"|json_encode }}\n{% endblock %}",
+            '1_include' => '{{ include("1_basic1", sandboxed=true) }}',
```
Comment (CarsonF) [inline test/Twig/Tests/Extension/SandboxTest.php]: Well that would throw `Twig_Error_Loader` and disabling the sandbox is already checked for that exception.
https://github.com/twigphp/twig/blob/1.x/lib/Twig/Extension/Core.php#L1402-L1405

Comment (fabpot) [inline test/Twig/Tests/Extension/SandboxTest.php]: nvm

**Comment by fabpot** [comment]
Thank you @CarsonF.

**Pull request merged**